wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WLabel.java

```diff
@@ -111,10 +111,11 @@ public String getText() {
 	 * @param args optional arguments for the message format string.
 	 */
 	public void setText(final String text, final Serializable... args) {
-		String currText = getText();
+		Serializable currText = getComponentModel().text;
+		Serializable textToBeSet = I18nUtilities.asMessage(text, args);
 
 		if (!Objects.equals(text, currText)) {
```

Comment:
Changes are looking good, except the equals test needs to be `textToBeSet` and `currText`.